dask/array/tests/test_reductions.py

```diff
@@ -566,3 +566,25 @@ def test_regres_3940(func):
     assert func(a).name != func(a + 1).name
     assert func(a, axis=0).name != func(a).name
     assert func(a, axis=0).name != func(a, axis=1).name
+
+
+def test_trace():
+    def _assert_eq(a, b, dtype=np.int64):
+        return assert_eq(a.compute(), np.asarray(b, dtype))
```

Comment:
We prefer to pass the array `a` into the object properly, rather than calling compute.  This allows us to run a variety of checks.

Reply:
That makes sense. Done!